### PR TITLE
3209 feat/export dataloggers and sensors

### PR DESCRIPTION
**Pull request by perryr16**
#### Any background context you want to provide?
This PR allows users to download the sensor and sensor reading data as a csv

#### How should this be manually tested?
Import sensor and sensor readings for a property using the following test files: 
[small_sensor_readings_example.csv](https://github.com/SEED-platform/seed/files/8532594/small_sensor_readings_example.csv)
[example-sensor-metadata.xlsx](https://github.com/SEED-platform/seed/files/8532595/example-sensor-metadata.xlsx)

Inventory > property (i) > sensors > create datalogger > upload sensors using "example-sensor-metadata.xlsx" > upload sensor readings using "small_sensor_readings_example.csv" > export as CSV for both sensors and readings

NOTE: see updated testing instructions in the comment below

#### What are the relevant tickets?
#3209 

#### Screenshots (if appropriate)

Sample generated CSVs
[Sensor_Usage_for_Property_3207429_2022_04_26.csv](https://github.com/SEED-platform/seed/files/8564953/Sensor_Usage_for_Property_3207429_2022_04_26.csv)
[Sensors_for_Property_3207429_2022_04_26.csv](https://github.com/SEED-platform/seed/files/8564954/Sensors_for_Property_3207429_2022_04_26.csv)

### CODE REVIEW

File: seed/static/seed/js/controllers/inventory_detail_sensors_controller.js [hunk 1]
```diff
@@ -175,6 +175,21 @@ angular.module('BE.seed.controller.inventory_detail_sensors', [])
         fastWatch: true
       };
 
+      $scope.exportSensorData = function (grid_data, data_type) {
+        let title = `${data_type}_for_${$scope.inventory_type == 'properties' ? 'Property' : 'Taxlot'}_${$scope.item_state.pm_property_id}_${moment().format('YYYY_MM_DD')}`;
+        let keys = grid_data.columnDefs.map(c => c.field);
+        let data = [keys.join(',')];
+
+        grid_data.data.forEach(d => {
+          let row = []
```
Comment (Ryoken) [inline seed/static/seed/js/controllers/inventory_detail_sensors_controller.js:184]: Could these three lines get semicolons?

**Review by Ryoken — APPROVED** [review]
Looks good after the tests pass!

**Comment by haneslinger** [comment]
This is looking pretty good!

I was under the impression that we wanted the exported file to look just like the import one, that is, like [example-sensor-metadata.xlsx](https://github.com/SEED-platform/seed/files/8532595/example-sensor-metadata.xlsx). Might be worth double checking with @nllong, but other than that is looks good!

**Comment by perryr16** [comment]
Seems the difference is the presence of the `data_logger` column. Also curious if the downloaded file should be formatted as .xlsx

File: seed/static/seed/js/controllers/inventory_detail_sensors_controller.js [hunk 2]
```diff
@@ -175,6 +175,41 @@ angular.module('BE.seed.controller.inventory_detail_sensors', [])
...
+      }
+      $scope.exportSensorUsages = function (grid_data, sensor_data, data_type) {
+        let title = `${data_type}_for_${$scope.inventory_type == 'properties' ? 'Property' : 'Taxlot'}_${$scope.item_state.pm_property_id}_${moment().format('YYYY_MM_DD')}`;
+        let keys = grid_data.columnDefs.map(c => c.field)
+
+        // grid data headers are a combination of display name and data logger. Use sensor_data to make a conversion object
+        let display_to_column = {'timestamp': 'timestamp'}
+        sensor_data.data.forEach(sensor => {
+          display_to_column[`${sensor.display_name} (${sensor.data_logger})`] = sensor.column_name
+        })
```
Comment (perryr16) [inline seed/static/seed/js/controllers/inventory_detail_sensors_controller.js:199]: To allow backward compatibility, the headers need to match `sensor.column_name`.

**Comment by perryr16** [comment]
## Updated Testing Procedure
After downloading sensors and sensor_readings csv, go to a different property, create a datalogger and upload the recently exported csv files as sensor and sensor_readings. This should populate sensors and sensor_readings with the same information

**Review by Ryoken — APPROVED** [review]
Works as expected.  Could the JS changes get semicolons?  Otherwise it looks good!

**Review by nllong — APPROVED** [review]
worked as expected!

image

image

I pushed a small change to the name of the test files so that `example` is prepended instead of appended and spinal-case @haneslinger. Just for consistency.

**Pull request merged**